objects/src/assets/mod.rs

```diff
@@ -202,15 +208,20 @@ impl TryFrom<&[u8; 32]> for Asset {
 
 impl Serializable for Asset {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
-        let data: [u8; 32] = self.into();
+        let data: [u8; Asset::SERIALIZED_SIZE] = self.into();
         target.write_bytes(&data);
     }
+
+    fn get_size_hint(&self) -> usize {
+        Asset::SERIALIZED_SIZE
+    }
```

Comment:
I think this is fine for now, but ideally, we should serialize fungible and non-fungible assets "natively" (i.e., we should have dedicated serializers/deserializes on `FungibleAsset` and `NonFungibleAsset`). Could you create an issue for this?

Reply:
Created https://github.com/0xPolygonMiden/miden-base/issues/900. `Serializable` is already implemented for `NonFungibleAsset` its just not used in `Asset` serialization yet, so I added a comment in the issue.